TypeError with TS@5.1.3: Property 'hitsPerPage' does not exist on type 'IntrinsicAttributes & PlainSearchParameters'

### 🐛 Current behavior

So this might be a bug with typescript, but it's the only library that broke when updating it so I'm going to create the bug report here.

When running with version `5.1.3` of `typescript` and `react-instantsearch-hooks-web` I get a type error on the `Configure` component, it appears it's no longer inferring the `SearchOptions` correctly. Downgrading to version `5.0.4` of `typescript` fixes the issue.

```tsx
> tsc --noEmit

components/SearchApp/SearchApp.tsx:28:18 - error TS2322: Type '{ hitsPerPage: number; filters: string | undefined; }' is not assignable to type 'IntrinsicAttributes & PlainSearchParameters'.
  Property 'hitsPerPage' does not exist on type 'IntrinsicAttributes & PlainSearchParameters'.

28       <Configure hitsPerPage={hitsPerPage} filters={filters} />
                    ~~~~~~~~~~~
```

### 🔍 Steps to reproduce

1. Install the following packages `react-instantsearch-hooks-web@6.44.1 typescript@5.1.3`
2. Create a component with
```tsx
<InstantSearch searchClient={searchClient} indexName={indexName}>
  <Configure hitsPerPage={hitsPerPage} filters={filters} />
  {children}
</InstantSearch>
```
4. Run `tsc --noEmit`

```
> tsc

components/SearchApp/SearchApp.tsx:28:18 - error TS2322: Type '{ hitsPerPage: number; filters: string | undefined; }' is not assignable to type 'IntrinsicAttributes & PlainSearchParameters'.
  Property 'hitsPerPage' does not exist on type 'IntrinsicAttributes & PlainSearchParameters'.

28       <Configure hitsPerPage={hitsPerPage} filters={filters} />
                    ~~~~~~~~~~~

```

Or checkout this repro https://stackblitz.com/edit/algolia-ts-bug-2mfs3u?file=app%2FSearchApp.tsx and run `npm run typecheck` in the terminal.

### Live reproduction

https://stackblitz.com/edit/algolia-ts-bug-2mfs3u?file=app%2FSearchApp.tsx

### 💭 Expected behavior

There should be no typescript error as with previous versions of typescript.

### Package version

react-instantsearch-hooks-web@6.44.1, typescript@5.1.3

### Operating system

Windows, Linux

### Browser

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/packages/instantsearch.js/package.json
+++ b/packages/instantsearch.js/package.json
@@ -33,7 +33,7 @@
     "@types/google.maps": "^3.45.3",
     "@types/hogan.js": "^3.0.0",
     "@types/qs": "^6.5.3",
-    "algoliasearch-helper": "^3.13.0",
+    "algoliasearch-helper": "^3.13.2",
     "hogan.js": "^3.0.2",
     "htm": "^3.0.0",
     "preact": "^10.10.0",

--- a/packages/react-instantsearch-core/package.json
+++ b/packages/react-instantsearch-core/package.json
@@ -38,7 +38,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.1.2",
-    "algoliasearch-helper": "^3.13.0",
+    "algoliasearch-helper": "^3.13.2",
     "prop-types": "^15.6.2",
     "react-fast-compare": "^3.0.0"
   },

--- a/packages/react-instantsearch-dom/package.json
+++ b/packages/react-instantsearch-dom/package.json
@@ -41,7 +41,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.1.2",
-    "algoliasearch-helper": "^3.13.0",
+    "algoliasearch-helper": "^3.13.2",
     "classnames": "^2.2.5",
     "prop-types": "^15.6.2",
     "react-fast-compare": "^3.0.0",

--- a/packages/react-instantsearch-hooks/package.json
+++ b/packages/react-instantsearch-hooks/package.json
@@ -49,7 +49,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.1.2",
-    "algoliasearch-helper": "^3.13.0",
+    "algoliasearch-helper": "^3.13.2",
     "instantsearch.js": "4.56.1",
     "use-sync-external-store": "^1.0.0"
   },

--- a/packages/vue-instantsearch/package.json
+++ b/packages/vue-instantsearch/package.json
@@ -60,7 +60,7 @@
     "@vue/test-utils": "1.3.0",
     "@vue/test-utils2": "npm:@vue/test-utils@2.0.0-rc.11",
     "algoliasearch": "4.14.3",
-    "algoliasearch-helper": "3.13.0",
+    "algoliasearch-helper": "3.13.2",
     "instantsearch.css": "8.0.0",
     "rollup": "1.32.1",
     "rollup-plugin-babel": "4.4.0",

--- a/tests/mocks/package.json
+++ b/tests/mocks/package.json
@@ -3,7 +3,7 @@
   "version": "1.14.0",
   "private": true,
   "dependencies": {
-    "algoliasearch-helper": "3.13.0",
+    "algoliasearch-helper": "3.13.2",
     "instantsearch.js": "4.56.1"
   }
 }

